### PR TITLE
added regex validation for names

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/main/java/com/vinyl/service/UserService.java [hunk 1]
```diff
@@ -1,8 +1,14 @@
 package com.vinyl.service;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+//import javax.ws.rs.BadRequestException;
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/UserService.java]: remove this line

File: src/main/java/com/vinyl/controller/UserController.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 		user.setEmail(userDTO.getEmail());
 		user.setPass(userDTO.getPass());
 
+
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/UserController.java]: unintended change

File: src/main/java/com/vinyl/service/UserService.java [hunk 3]
```diff
@@ -28,6 +34,14 @@ public User addUser(User user) {
 		user.setAddress("Defaul Address");
 		user.setRole(r);
 
+		Pattern p = Pattern.compile("^[A-Z][a-zA-Z]+$");
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/UserService.java]: create a String constant for the pattern "^[A-Z][a-zA-Z]+$"

Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/UserService.java]: move the validation to a new class that will handle only validation

File: src/main/java/com/vinyl/service/exception/BadRequestException.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.vinyl.service.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@SuppressWarnings("serial")
+@ResponseStatus(value= HttpStatus.BAD_REQUEST)
+public class BadRequestException extends RuntimeException{
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/exception/BadRequestException.java]: good job finding this solution

Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/exception/BadRequestException.java]: thank you!

File: src/main/java/com/vinyl/controller/UserController.java (hunk 2 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/UserController.java]: done

File: src/main/java/com/vinyl/service/UserService.java (hunk 3 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/UserService.java]: done

File: src/main/java/com/vinyl/service/UserService.java (hunk 1 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/UserService.java]: done

File: src/main/java/com/vinyl/service/UserService.java (hunk 3 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/UserService.java]: done

File: src/main/java/com/vinyl/service/Validation.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.vinyl.model.User;
+import com.vinyl.service.exception.BadRequestException;
+
+public class Validation {
+
+	public static final String nameRegex = "^[A-Z][a-zA-Z]+$";
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/Validation.java]: it is not used outside, make it private

Comment (dragosslujeru) [reply src/main/java/com/vinyl/service/Validation.java]: use constant naming conventions

File: src/main/java/com/vinyl/service/Validation.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+	public static final Pattern VALID_NAME_REGEX = Pattern.compile(nameRegex);
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/Validation.java]: it is a Pattern, not a regex VALID_NAME_PATTERN

File: src/main/java/com/vinyl/service/Validation.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package com.vinyl.service;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.vinyl.model.User;
+import com.vinyl.service.exception.BadRequestException;
+
+public class Validation {
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/Validation.java]: Validator

Comment (dragosslujeru) [reply src/main/java/com/vinyl/service/Validation.java]: make it a spring bean and inject into the service

File: src/main/java/com/vinyl/service/Validation.java (hunk 5 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/Validation.java]: fixed

File: src/main/java/com/vinyl/service/Validation.java (hunk 6 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/Validation.java]: fixed

File: src/main/java/com/vinyl/service/Validation.java (hunk 7 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/Validation.java]: fixed

**Review by dragosslujeru — APPROVED** [review]

**Pull request merged**